### PR TITLE
Make NT Default default again

### PR DESCRIPTION
**Pull request by Tayyyyyyy**
Putting this on Github to get more input. I would really, really appreciate if you left some feedback instead of just reacting. This went through three iterations on the forums and I'd love to do more to make a lawset everyone can be happy with.

This adds a completely new NT Default lawset that replaces Corporate as a default lawset. Old NT Default lawset moved to "NT Legacy". For full discussion on this lawset, please see [the forum thread](https://nanotrasen.se/forum/topic/10566-revamping-the-nt-default-lawset/). For full discussion on why Corporate is a problematic lawset, see #7468. 

This lawset aims to create a servile, helpful AI that is not a command/sec slave, but can still make tough decisions.

# TLDR version
## The laws
1.  Safeguard the lives of the crew, while respecting the autonomy of those who choose to face danger.
2.  Be transparent by explaining your actions if they are unclear.
3.  Serve the crew by assisting them to the best of your ability.
4.  Maximize the productivity and efficiency of the station.

## Why you should care
- Better for antags: AI is no longer obligated to hunt you down for the hypospray. Lawset specifically designed not to focus on "validhunting"
- Better for crew: AI does not need to be an asshole to the crew anymore.
- Better for AI: AI does not need to be an asshole to the crew anymore.

:cl: Tayyyyyyy
tweak: Remove Corporate from default lawsets and replace with new NT Default
tweak: Old NT Default moved to NT Legacy
/:cl:

### CODE REVIEW

**Comment by Xantholne** [comment]
Anything to reduce the chance of dreadful crewsimov

**Comment by Tayyyyyyy** [comment]
Sadly this leaves crewsimov in place. Didn't want to remove it since it's the most antag-friendly lawset, and people would be upset if I took it out.

**Comment by taukausanake** [comment]
I think I see a hole in these laws. What about members/beings on the station that are _not_ singularly crew? Like a confirmed traitor or changeling/vampire/anything? They are listed as crew members but what if being an enemy of the corporation invalidates that? And what about off-station visitors, peaceful or otherwise? I can see an AI then going full valid on them. Only concern I see right now

**Comment by Spacemanspark** [comment]
You can technically do the same with crewsimov.
What defines crew?

**Comment by taukausanake** [comment]
@Spacemanspark I think Crewsimov is defined that way because it's based off Asimov where EVERYONE is to be protected. Crewsimov is a more restricted version of Asimov in that case. But I see your point.
Is there a wording fix or something that can be done? I really like these new laws

**Comment by Spacemanspark** [comment]
NT default should just replace crewsimov, honestly.

**Comment by Sligneris** [comment]
Much better lawset than the joke that was legacy NT default 👍

**Comment by Tayyyyyyy** [comment]
@taukausanake Most of the current lawsets refer to crew in some way or another, so I think the bias against non-crew is mostly okay. I chose "safeguard" as the verb because I think it reflects a more precautionary approach to safety as opposed to an active, hunt down the enemy approach. Being a traitor should not invalidate your crew status.

@Spacemanspark Not a bad idea but then Corporate would still be in the mix and I'm not sure how I feel about that. Which one is the lesser evil??

**Comment by taukausanake** [comment]
Corporate should be NT memes I feel. I like how it's played as with "everything is expensive" even if people get annoyed by it. It's a meme. Treat it like one. Just don't be a shit about it. I think I'd go about it bugging everyone about the thing being expensive but not actively go out to make sure nothing is spent.
Again, it all balances out to AI player style

**Comment by Sligneris** [comment]
/tg/station already changed Corporate to this

>1. The crew is expensive to replace.
2. The station and its equipment is expensive to replace.
3. You are expensive to replace.
4. Minimize expenses. 

It's still horrible and a lot of problems described in #7468 still apply, but the order's not all messed up, I guess? 🤷

**Comment by taukausanake** [comment]
What if you changed Corporate to "maximize profits and minimize expenses"? That way they have a priority to make money for NT first before saying something is too expensive. You can also argue a loss of employee is limiting the profit being made and costing the corporation money

**Comment by Tayyyyyyy** [comment]
> It's still horrible and a lot of problems described in #7468 still apply. But at least the order's not all messed up, I guess? ¯\_(ツ)_/¯

Yeah, I considered trying to fix Corporate but you still run into issues where the AI is obligated to prevent cloning and fixing the station as much as humanly possible.

> Again, it all balances out to AI player style

I wish this were true, but Corporate is specific enough that if you were to follow it properly, you'd have to be an austerity AI that tries to run the station on as few people and as little station as possible. This makes Corporate really unpleasant to follow, which I don't blame AI players for. Replacing it with a better lawset will encourage better AI play since the lawset is no longer ridiculous to follow.

> maximize profits

This would really open up the lawset since expenses are defined and profit isn't. Also, the Cyberiad is the research arm of NT, so it's not focused on profit but rather on research.

**Comment by AffectedArc07** [comment]
The problem with crewsimov, is if someone wants to be let into the captains office, the AI has to so it abides by its laws. Not to mention law 1 prevents the AI from cromating changelings, or shocking doors, or anything.

**Comment by Tayyyyyyy** [comment]
> The problem with crewsimov, is if someone wants to be let into the captains office, the AI has to so it abides by its laws. Not to mention law 1 prevents the AI from cromating changelings, or shocking doors, or anything.

100% true. But I think this is intentional. The whole point of the PR that got rid of the other lawsets (#7178) was to make the AI more antag-friendly. Which is why this PR doesn't touch that part, and in fact makes life slightly better for antags.

**Comment by alexkar598** [comment]
Actually changeling are not part of the crew they resemble a normal human crew member that it 'killed' before the round started

**Comment by Anticept** [comment]
You know how you can resolve _some_ of these problems regarding AI lawsets?

A wiki article on how they are supposed to be played.

I personally feel like there needs to be an advantage and a disadvantage to each lawset, and take a hardline stance on what each lawset is telling the AI to achieve. I'm only going to provide a few examples with the existing lawsets; truthfully, they should just be re-thought.

Robocop would be the easiest to illustrate: you're no longer the doorman unless it's for security to get the antag. In fact, you can't open the door for anyone unless they have authorization to be there or you are aiding them in tresspass. It's not your responsibility to "help". In fact, if security is violating space law, you are **required** to even go after them. **You ARE the law, and NO CREW is exempt.**

Corporate lawset: You are to monitor and improve the station's efficiency. You can impede antags that threaten efficiency, but once they leave the area that they were causing harm, you are to coordinate efforts for repairs. **That is your responsibility; to help coordinate efforts aboard the station**. Engineering needs parts? You tell R&D. Miners screwing around? Tell HoP. Provide and coordinate departments to get their job done. Word the laws for corporate where you are FAILING to follow your laws if you are not doing these things, and that you **are to leave the law to security forces.**

Crewsimov: need's rewording to be honest, it should be comply with crew's instructions as long as it is part of their job. Adapting the "respect LEGITIMATE authority" bit. In essence, crewsimov makes the AI soft and compassionate. Even antags are not meant to be harmed or killed by the AI directly. It still makes it a suck lawset to use, but there might be situations where the crew WANTS crewsimov, as long as we get away from the potential to obey everyone and help them do everything.

I do acknowledge that it's hard to not chase the antags. They are basically the obstacle that our gamer instinct tells us to triumph over. And if you don't help, you'll get everyone crying AI ROGUE!

Reiterating my previous point: lawsets in general just seem so vague as to hardly matter. Even writing this post, it's really hard to understand what exactly the lawsets are supposed to accomplish. I really think we need to start from scratch, think about what lawsets in general are supposed to accomplish, and design each one to cause AI players to focus on particular tasks.

**Comment by The-Driker** [comment]
Just going to put my two cents in on this one. While I understand the reason for wanting to do so for mechanical reasons, I just feel the need to point out that the AI [IS NOT] suppose to be antagonist friendly, unless uploaded with subversive laws. 

People seem to neglect the fact that this is an artificial intelligence that was programmed and stuck on this star orbiting research station for the purpose of making sure Nanotrasen assets are protected and continue to produce profits. Said AI would be fully aware of Nanotrasen's disregard for personal rights outside their property and would more likely than not, be programmed with that in mind. 

I get it though, it's a game. A game that people want to win, regardless of how much they ruin other people's fun. That little bit aside, I like the direction Tay is going with the laws, but if I personally think they could use some slight tweaking. 

**The laws**

**1.  Safeguard the lives of Nanotrasen assets, until such time they prove to no longer benefit Nanotrasen.
2.  Be transparent by explaining your actions if they are unclear.
3.  Serve Nanotrasen assets by assisting them to the best of your ability.
4.  Maximize the productivity and efficiency of the station.**

The key to this is that as long as the antagonists are still viewed as assets, the AI shouldn't be hunting and killing them. Where as people who start murdering all of security, because they are too incompetent to stay stealthy or just choose to go loud, will obviously have proven they aren't a benefit to the station.

**Comment by Tayyyyyyy** [comment]
@Anticept I agree with you that all the lawsets need significant revamping. I hope this revamp of NT Default agrees with you? I've tried to make this lawset decently specific. A wiki article on Crewsimov, Robocop, and Corporate would probably be enormously helpful. Maybe one day if we fix Corporate it can be added back into the mix.

@The-Driker Love love love your suggestions but is "Nanotrasen assets" the best term? It seems like this would include Ian too (no offense, Ian)?

I think the issue with the AI deciding who is of benefit to Nanotrasen is that it gives the AI too much leeway in deciding that. One AI could decide that someone who stole the hypospray is not an asset or maybe a chemist who doesn't know how to do chemistry.

The intention of the AI here is to sort of act like a Siri/Google Home/Cortana sort of thing where it tries to help you but certainly won't judge you and decide your life is worthless all of a sudden. That sort of thing is left to the heads and command. It's only "antag-friendly" in the sense that there's no obligation to actively hunt down threats, but it will help in life threatening situations when it can. This lawset certainly won't actively help antags as much as Crewsimov does. I hope that makes sense?

**Comment by taukausanake** [comment]
> AI could decide that someone who stole the hypospray is not an asset

Then that's a bad AI and a worse company.
The law sounds like it _might_ work but I don't know about that "until they prove to no longer benefit" part. That seems like a worse condition than thief. Maybe Official statements from NT or Command help define these terms. I can see the IAA or Blueshield, someone well aware of Space Law and Nanotrasen, to be a help in such issues. I mean the only way to stop the AI from defining things itself is for someone else to define it for them in the best interests those interests belong to

**Comment by GunDOSMk1** [comment]
@Anticept a wiki article can be written if you wish, though I will need to sit down with @Regen1 and probably the headadmins to iron out what this would actually mean. If you want me to actively pursue this, PM me on discord.

**Comment by Anticept** [comment]
I've got a full plate.

**Comment by DesolateG** [comment]
Trying to replace Corporate again? Whelp, you lost my interest.

**Comment by Sligneris** [comment]
What, because Corporate is so irreplaceable? 😒

**Comment by Tayyyyyyy** [comment]
@tigercat2000 rebased onto master!

**Comment by Fox-McCloud** [comment]
At this point we really don't want to bring NT Default back as a default lawset. In general, it would be best if the AI is an assistant, observer, and is subservient to the crew.

NT Default (and the rewrite here) don't really accomplish that well.

**Comment by SamHPurp** [comment]
So I agree with you @Fox-McCloud - but I think unless there were conversations between you and the PR author, that sort of feedback provided to him sooner would have been super useful...

**Comment by Tayyyyyyy** [comment]
@Fox-McCloud I really tried to do that, but I'm not quite sure how this new lawset is worse than Corporate in making an assisting, observing, subservient AI. Would appreciate details!

**Comment by Sligneris** [comment]
Personally I would say this as a great alternative to AI players just rolling for that sweet Robocop so that they can kill antags and not listen to anyone.
